sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -36,12 +36,12 @@ abstract sealed class NullOrdering {
 
 case object Ascending extends SortDirection {
   override def sql: String = "ASC"
-  override def defaultNullOrdering: NullOrdering = NullsFirst
+  override def defaultNullOrdering: NullOrdering = NullsLast
 }
 
 case object Descending extends SortDirection {
   override def sql: String = "DESC"
-  override def defaultNullOrdering: NullOrdering = NullsLast
+  override def defaultNullOrdering: NullOrdering = NullsFirst
```

Comment:
I think this is a breaking change and workaround looks pretty easy. Can you check what other DBMSs do? If it's consistent with some other DBMSs, I think we shouldn't fix it and resolve the ticket as `Won't Do`

Reply:
@HyukjinKwon 
From the below link,  spark is consistent with **mysql, sqlserver**
But inconsitent with **DB2, Oracle, Postgresql**
https://docs.mendix.com/refguide/null-ordering-behavior#3-overview-of-default-nulls-sort-order

![image](https://user-images.githubusercontent.com/7912929/61195812-3dbe0e80-a6e8-11e9-90e1-bb496843a2bd.png)

Reply:
I think the current behaviour already makes sense. I wouldn't fix it. cc @dongjoon-hyun, @wangyum, @gatorsmile WDYT?

Reply:
+1 for @HyukjinKwon 's comment.